run_all_experiments.py

```diff
@@ -296,14 +300,42 @@ def _print_experiment_result(result: Result):
         f'{result.result}')
 
 
-def _print_experiment_results(results: list[Result]):
+def _print_experiment_results(results: list[Result], cov_gain: Dict[str, float]):
```

Comment:
For typing, could you follow https://peps.python.org/pep-0585/ -- i.e. `dict`, `list`, `tuple` etc. (full list in the pep) instead of `Dict`, `List`, `Tuple` etc.